### PR TITLE
Added custom radii for each corner of RRect.

### PR DESCRIPTION
**Pull request by dragostis**
In order to implement more detailed animation in Flutter, precise
handling of the rounded corner's radii is necessary. This commit
adds a data structure Radius and updated RRect accordingly. [flutter/engine PR](https://github.com/flutter/engine/pull/2820).

### CODE REVIEW

File: packages/flutter/lib/src/material/checkbox.dart [hunk 1]
```diff
@@ -159,7 +159,7 @@ class _RenderCheckbox extends RenderToggleable {
     double rectSize = _kEdgeSize - inset * _kStrokeWidth;
     Rect rect = new Rect.fromLTWH(offsetX + inset, offsetY + inset, rectSize, rectSize);
 
-    RRect outer = new RRect.fromRectXY(rect, _kEdgeRadius, _kEdgeRadius);
+    RRect outer = new RRect.fromRectAndRadius(rect, const Radius.circular(_kEdgeRadius));
```
Comment (Hixie) [inline packages/flutter/lib/src/material/checkbox.dart]: Make kEdgeRadius and kTrackRadius (in switch.dart) into Radius constants

**Comment by Hixie** [comment]
The part you've posted so far is LGTM (once you've revved the engine, and modulo the comments above). Please do the other changes in separate PRs for ease of review.

**Comment by dragostis** [comment]
@Hixie Sure. I was thinking about merging PRs to this PR.

File: packages/flutter/lib/src/material/checkbox.dart (hunk 1 above)
Comment (dragostis) [inline packages/flutter/lib/src/material/checkbox.dart]: `kTrackRadius` is used in a calculation below. Maybe we should keep it a `double`?

Comment (Hixie) [inline packages/flutter/lib/src/material/checkbox.dart]: Sure, that's fine.

**Comment by Hixie** [comment]
Make sure this patch includes the engine rev

File: packages/flutter/test/ui/rrect_test.dart [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+  test("RRect.contains()", () {
+    RRect rrect = new RRect.fromRectCustom(
+      new Rect.fromLTRB(1.0, 1.0, 2.0, 2.0),
+      new Radius.circular(0.5),
+      new Radius.circular(0.25),
+      new Radius.elliptical(0.25, 0.75),
+      Radius.zero
+    );
+
+    expect(rrect.contains(new Point(1.0, 1.0)), equals(false));
```
Comment (Hixie) [inline packages/flutter/test/ui/rrect_test.dart]: you can do isFalse instead of equals(false). Same with isTrue.

Comment (Hixie) [inline packages/flutter/test/ui/rrect_test.dart]: also, use const Point instead of new Point

File: packages/flutter/test/ui/rrect_test.dart [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/painting.dart';
+
+import 'package:test/test.dart';
+
+void main() {
+  test("RRect.contains()", () {
+    RRect rrect = new RRect.fromRectCustom(
```
Comment (Hixie) [inline packages/flutter/test/ui/rrect_test.dart]: yeah, looking at this I really think we need a better name than fromRectCustom.

Maybe fromRectAndCorners and then use named arguments for the corners?

**Pull request closed without merging**